doc/handle-ASIC-SDK-health-event/handle-ASIC-SDK-health-event.md

```diff
@@ -0,0 +1,635 @@
+  - Capabilities will be fetched ahead of registering the event and exposed to `STATE_DB`.
+  - A user can suppress the events that he/she is not interested in by severity and category using configuration.
+- A vendor SAI notifies orchagent an ASIC/SDK event using `switch_asic_sdk_health_event` notification with corresponding arguments when it detects an HW/SW issue.
+  - The orchagent stores the information of the ASIC/SDK health event into database and pushes it to gNMI server using event collector mechanism.
+- The vendor SAI notifies orchagent to shutdown using `switch_shutdown_request` notification if the event is seriously enough.
+    Orchagent will abort on receiving the notification.
+  - Otherwise, `switch_shutdown_request` will not be sent and system continues to run.
+- The ASIC/SDK health events stored in `STATE_DB.ASIC_SDK_HEALTH_EVENT_TABLE` can be displayed or cleared using CLI commands.
+
+The `timestamp`, `severity`, and `category` are represented in various components.
+
+The `timestamp` is converted to format "%Y-%m-%d %H:%M:%S" which is a walltime based on the timezone in `swss` docker container.
+
+The `severity` is mapped between each other according to the next table:
+
+| represention in SONiC | Enumerate in SAI headers | SAI attribute to register corresponding eventa |
+|:---:|:---:|:---:|
+| fatal | SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_FATAL | SAI_SWITCH_ATTR_REG_FATAL_SWITCH_ASIC_SDK_HEALTH_CATEGORY |
+| warning | SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_WARNING | SAI_SWITCH_ATTR_REG_WARNING_SWITCH_ASIC_SDK_HEALTH_CATEGORY |
+| notice | SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_NOTICE | SAI_SWITCH_ATTR_REG_NOTICE_SWITCH_ASIC_SDK_HEALTH_CATEGORY |
+
+The `category` is mapped between each other according to the next table:
+
+| represention in SONiC | Enumerate in SAI headers |
+|:---:|:---:|
+| software | SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_SW |
+| firmware | SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_FW |
+| cpu_hw | SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_CPU_HW |
+| asic_hw | SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_ASIC_HW |
+
+This is a built-in SONiC feature implemented in the following sub-modules
+
+- sonic-swss, which handles SAI notification, storing it into database and pushing it into gNMI server
+- sonic-sairedis, which transmits the ASIC/SDK health events reported by vendor SAI to orchagent
+- sonic-utilities, in which the CLI to display and clear ASIC/SDK health events and configure suppress ASIC/SDK health events are implemented
+- sonic-buildimage, in which the new yang models for the new events are defined
+
+#### DB changes
+
+##### STATE_DB change
+
+###### Table ASIC_SDK_HEALTH_EVENT_TABLE
+
+Table `ASIC_SDK_HEALTH_EVENT_TABLE` contains the ASIC/SDK health events information.
+
+```text
+key           =  ASIC_SDK_HEALTH_EVENT_TABLE:timestamp_string       ; "%Y-%m-%d %H:%M:%S", full-date and partial-time separated by white space.
```

Comment:
Does it need the ASIC no as part of the key in case of multi-asic use case?

Reply:
AFAIU, `STATE_DB` is per-ASIC on multi ASIC platform.
Do we still need the ASIC name in it?